### PR TITLE
Tidy up some issues raised by linters and type checkers

### PR DESCRIPTION
**Pull request by erbridge**

### CODE REVIEW

File: python/cog/director/__main__.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def _die(signum: Any, frame: Any) -> None:
 
 args = parser.parse_args()
 
-events = queue.Queue(maxsize=128)
+events: queue.Queue = queue.Queue(maxsize=128)
```
Comment (nickstenning) [inline python/cog/director/__main__.py:50]: lol, so glad that the python typing ecosystem has such powerful type inference :P

File: python/cog/server/runner.py [hunk 2]
```diff
@@ -72,7 +69,7 @@ def setup(self) -> Tuple[schema.Status, str]:
     # no longer have to support Python 3.8
     def predict(
         self, prediction: schema.PredictionRequest, upload: bool = True
-    ) -> AsyncResult:
```
Comment (erbridge) [inline python/cog/server/runner.py:75]: Note this is a change, not just additive.

File: python/cog/director/director.py [hunk 3]
```diff
@@ -371,15 +362,15 @@ def _record_failure(self):
                 "saw too many failures in a row", failure_count=self._failure_count
             )
 
-    def _record_success(self):
+    def _record_success(self) -> None:
         self._failure_count = 0
 
-    def _abort(self, message=None, **kwds):
+    def _abort(self, message: str = None, **kwds: Any) -> None:
```
Comment (nickstenning) [inline python/cog/director/director.py:368]: I'm not quite sure why `message` was optional before but I don't think it needs to be or should be.
```suggestion
    def _abort(self, message: str, **kwds: Any) -> None:
```

File: python/cog/director/__main__.py (hunk 1 above)
Comment (erbridge) [reply python/cog/director/__main__.py:50]: Seems to have problems with `queue.Queue` but basically nothing else 🤷

**Review by nickstenning — APPROVED** [review]
LGTM.

**Pull request merged**